### PR TITLE
Add doc for new function pointer support

### PR DESCRIPTION
**Pull request by steveharter**
Adds new doc for the new [APIs](https://github.com/dotnet/runtime/issues/69273) added for .NET 8.0 Preview 2.

### CODE REVIEW

**Comment by steveharter** [comment]
@docs this is for .NET 8.0 (Preview 2). The Preview URL doesn't yet support .NET 8.0 -- any way to verify the changes?

**Comment by carlossanlop** [comment]
Is this ready to review?

**Comment by steveharter** [comment]
>Is this ready to review?

Yes, thanks.

File: xml/System.Reflection/FieldInfo.xml [hunk 1]
```diff
@@ -660,9 +660,23 @@
...
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Gets the modified type of this field object.</summary>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+ A modified type supports <xref:System.Type.GetOptionalCustomModifiers%2A>, <xref:System.Type.GetRequiredCustomModifiers%2A> and <xref:System.Type.GetFunctionPointerCallingConventions%2A> which are used to to obtain custom modifiers from a function pointer.
```
Comment (buyaa-n) [inline xml/System.Reflection/FieldInfo.xml:668]:
```suggestion
 A modified type supports <xref:System.Type.GetOptionalCustomModifiers%2A>, <xref:System.Type.GetRequiredCustomModifiers%2A> and <xref:System.Type.GetFunctionPointerCallingConventions%2A> which are used to obtain custom modifiers from a function pointer.
```

File: xml/System.Reflection/ParameterInfo.xml [hunk 2]
```diff
@@ -726,9 +726,23 @@
...
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Gets the modified type of this parameter object.</summary>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+ A modified type supports <xref:System.Type.GetOptionalCustomModifiers%2A>, <xref:System.Type.GetRequiredCustomModifiers%2A> and <xref:System.Type.GetFunctionPointerCallingConventions%2A> which are used to to obtain custom modifiers from a function pointer.
```
Comment (buyaa-n) [inline xml/System.Reflection/ParameterInfo.xml:734]:
```suggestion
 A modified type supports <xref:System.Type.GetOptionalCustomModifiers%2A>, <xref:System.Type.GetRequiredCustomModifiers%2A> and <xref:System.Type.GetFunctionPointerCallingConventions%2A> which are used to obtain custom modifiers from a function pointer.
```

File: xml/System.Reflection/PropertyInfo.xml [hunk 3]
```diff
@@ -989,9 +989,23 @@
...
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Gets the modified type of this property object.</summary>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+ A modified type supports <xref:System.Type.GetOptionalCustomModifiers%2A>, <xref:System.Type.GetRequiredCustomModifiers%2A> and <xref:System.Type.GetFunctionPointerCallingConventions%2A> which are used to to obtain custom modifiers from a function pointer.
```
Comment (buyaa-n) [inline xml/System.Reflection/PropertyInfo.xml:997]:
```suggestion
 A modified type supports <xref:System.Type.GetOptionalCustomModifiers%2A>, <xref:System.Type.GetRequiredCustomModifiers%2A> and <xref:System.Type.GetFunctionPointerCallingConventions%2A> which are used to obtain custom modifiers from a function pointer.
```

File: xml/System/Type.xml [hunk 4]
```diff
@@ -4209,9 +4209,26 @@ In .NET 6 and earlier versions, the <xref:System.Type.GetFields%2A> method does
...
+        </returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+ This method is provided for designers of managed compilers. For more information on custom modifiers, see classes in the <xref:System.Runtime.CompilerServices> namespace.  
+  
+ ]]></format>
+        </remarks>      
+        <exception cref="T:System.InvalidOperationException">The current <see cref="T:System.Type" /> object is not a function pointer. That is, the <see cref="P:System.Type.IsFunctionPointer" /> property returns <see langword="false" />.</exception>
```
Comment (buyaa-n) [inline xml/System/Type.xml:4231]: NIT: It seems to me better to start with `When` or `If`, up to you
```suggestion
        <exception cref="T:System.InvalidOperationException">If the current <see cref="T:System.Type" /> object is not a function pointer. That is, the <see cref="P:System.Type.IsFunctionPointer" /> property returns <see langword="false" />.</exception>
```

File: xml/System/Type.xml [hunk 5]
```diff
@@ -4237,9 +4254,14 @@ In .NET 6 and earlier versions, the <xref:System.Type.GetFields%2A> method does
...
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>When overridden in a derived class, returns the parameter types of the current function pointer <see cref="T:System.Type" />.</summary>
+        <returns>An array of <see cref="T:System.Type" /> objects representing all the parameter types for the current function pointer <see cref="T:System.Type" />.
+  
+ -or-  
+  
+ An empty array of type <see cref="T:System.Type" />, if no parameters are defined for the current function pointer <see cref="T:System.Type" />.
+        </returns>
+        <exception cref="T:System.InvalidOperationException">The current <see cref="T:System.Type" /> object is not a function pointer. That is, the <see cref="P:System.Type.IsFunctionPointer" /> property returns <see langword="false" />.</exception>
```
Comment (buyaa-n) [inline xml/System/Type.xml:4264]: NIT: Same here: 
```suggestion
        <exception cref="T:System.InvalidOperationException">If the current <see cref="T:System.Type" /> object is not a function pointer. That is, the <see cref="P:System.Type.IsFunctionPointer" /> property returns <see langword="false" />.</exception>
```

File: xml/System/Type.xml [hunk 6]
```diff
@@ -4265,9 +4287,9 @@ In .NET 6 and earlier versions, the <xref:System.Type.GetFields%2A> method does
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>When overridden in a derived class, returns the return type of the current function pointer <see cref="T:System.Type" />.</summary>
+        <returns>A <see cref="T:System.Type" /> object representing the return type for the current function pointer <see cref="T:System.Type" />.</returns>
+        <exception cref="T:System.InvalidOperationException">The current <see cref="T:System.Type" /> object is not a function pointer. That is, the <see cref="P:System.Type.IsFunctionPointer" /> property returns <see langword="false" />.</exception>
```
Comment (buyaa-n) [inline xml/System/Type.xml:4292]: NIT: same
```suggestion
        <exception cref="T:System.InvalidOperationException">If the current <see cref="T:System.Type" /> object is not a function pointer. That is, the <see cref="P:System.Type.IsFunctionPointer" /> property returns <see langword="false" />.</exception>
```

**Review by buyaa-n — APPROVED** [review]
Left some NIT, overall LGTM

File: xml/System/Type.xml (hunk 4 above)
Comment (steveharter) [reply xml/System/Type.xml:4231]: The existing usages of "InvalidOperationException" in similar cases don't use "If" so I kept that however the link to Type here is somewhat redundant so I removed it.

**Pull request merged**